### PR TITLE
Dagger android

### PR DESCRIPTION
**Pull request by VEndymionV**

### CODE REVIEW

File: app/src/main/java/com/example/training1/di/MoviesModule.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    fun provideMoviesRepository(): MoviesRepository {
+        return MoviesRepository(
+            provideRetrofitMoviesService(provideRetrofit()),
+            BuildConfig.API_KEY
+        )
+    }
+
+    @Singleton
+    @Provides
+    fun provideRetrofit(): Retrofit {
```
Comment (Bartosz-Kozajda) [inline app/src/main/java/com/example/training1/di/MoviesModule.kt]: this is not related to movies, should be in app component

File: app/src/main/java/com/example/training1/di/MoviesModule.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import javax.inject.Singleton
+
+const val API_URI = "http://www.omdbapi.com/"
+
+@Module
+class MoviesModule {
+
+    @Singleton
+    @Provides
+    fun provideMoviesRepository(): MoviesRepository {
```
Comment (Bartosz-Kozajda) [inline app/src/main/java/com/example/training1/di/MoviesModule.kt]: why this has to be Singleton? I think it should be in activity scope

File: app/src/main/java/com/example/training1/viewmodel/MoviesViewModel.kt [hunk 3]
```diff
@@ -22,7 +22,7 @@ class MoviesViewModel : ViewModel() {
     }
 
     private fun prepareDagger() {
```
Comment (Bartosz-Kozajda) [inline app/src/main/java/com/example/training1/viewmodel/MoviesViewModel.kt]: remove all dagger related stuff from viewmodel

File: app/src/main/java/com/example/training1/di/MoviesComponent.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import com.example.training1.model.MoviesRepository
+import com.example.training1.viewmodel.MoviesViewModel
+import com.example.training1.viewmodel.retrofit.MoviesService
+import dagger.Component
+import retrofit2.Retrofit
+import javax.inject.Singleton
+
+@Singleton
+@Component(modules = [MoviesModule::class])
+interface MoviesComponent {
```
Comment (Bartosz-Kozajda) [inline app/src/main/java/com/example/training1/di/MoviesComponent.kt]: read about @ContributesAndroidInjector annotation to generate components

File: app/src/main/java/com/example/training1/viewmodel/MoviesViewModel.kt (hunk 3 above)
Comment (VEndymionV) [reply app/src/main/java/com/example/training1/viewmodel/MoviesViewModel.kt]: Removed: https://github.com/VEndymionV/Training1/blob/dagger-android/app/src/main/java/com/example/training1/viewmodel/MoviesViewModel.kt

File: app/src/main/java/com/example/training1/di/MoviesComponent.kt (hunk 4 above)
Comment (VEndymionV) [reply app/src/main/java/com/example/training1/di/MoviesComponent.kt]: I used it in latest commit: https://github.com/VEndymionV/Training1/pull/17/commits/cb4f2daaa032ba057a9b6906b80a858f6561e7a8
